modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/TransactionsComponent.java

```diff
@@ -54,9 +57,31 @@ public void updateResourceConfiguration(ConfigurationUpdateReport report) {
             //a validation error on the server.
             configDef.getPropertyDefinitions().remove("process-id-socket-binding");
             config.remove("process-id-socket-binding");
+            //Also do not send updates for [process-id-socket-max-ports] because EAP expects it to
+            //be undefined and will assign 10 anyway.
+            configDef.getPropertyDefinitions().remove("process-id-socket-max-ports");
+            config.remove("process-id-socket-max-ports");
+        } else {
+            //EAP7 requires to unset [process-id-uuid] to allow setting [process-id-socket-binding]
+            config.getSimple("process-id-uuid").setValue(null);
         }
 
         ConfigurationWriteDelegate delegate = new ConfigurationWriteDelegate(configDef, getASConnection(), address);
         delegate.updateResourceConfiguration(report);
     }
+
+    @Override
+    public Configuration loadResourceConfiguration() throws Exception {
+        Configuration config = super.loadResourceConfiguration();
+
+        boolean processIdUuid = Boolean.valueOf(config.getSimpleValue("process-id-uuid"));
+        if (processIdUuid) {
+            // Do not send the value of [process-id-socket-max-ports] as this should be 10 and could confuse the user
+            // when trying to change the value.
+            config.remove("process-id-socket-max-ports");
+        }
+
+        return config;
+    }
```

Comment:
When [process-id-uuid] is true, [process-id-socket-max-ports] should be undefined  (according to the attribute description), but the value by default is '10'. 

I'm not sure if we should *hide* in the UI the value of [process-id-socket-max-ports] or if this would generate more confusion.